valhalla/midgard/sequence.h

```diff
@@ -321,6 +321,48 @@ template <class T> class sequence {
     std::sort(static_cast<T*>(memmap), static_cast<T*>(memmap) + memmap.size(), predicate);
     return;
   }
+  
+  // sort the file based on the predicate, and outputs to output_seq
+  //
+  // Strategy is to first sort sub-ranges of length buffer_size in place. 
+  // These should all fit in memory. Then, merge the sub-ranges into the 
+  // output sequence via priority queue.
+  void merge_sort(const std::function<bool(const T&, const T&)>& predicate,
+                  sequence<T>& output_seq,
```

Comment:
this is the only unfortunate bit, we have to keep around two sequences to make the implementation trivial... which means we double the disk need for a small period of time. i think its a worthwhile trade off for sure. at first i started thinking about a scheme where, before we sort the subsequences we shift the items on the beginning to the end fo the sequence to make space to swap things in from the priority queue as we do the merge but... that wont work you can think of pathological cases where the first "subsorted bucket" is all of the elements that sort to the end of the sequence, which means as we do the merge the space we made at the beginning will eventually eat into them. you cant win them all i guess :smile: